pandas/core/dtypes/cast.py

```diff
@@ -1044,7 +1044,7 @@ def maybe_cast_to_datetime(value, dtype, errors="raise"):
                     value = [value]
 
             elif is_timedelta64 and not is_dtype_equal(dtype, _TD_DTYPE):
-                if dtype.name in ("timedelta64", "timedelta64[ns]"):
+                if (dtype.kind == "m") and (dtype.name != "timedelta64[ps]"):
```

Comment:
let's leave [ps] for another issue. I am not sure what we should do about it. we *could* convert it with precision loss rather than actually raising (which is what I think it would do now)

Reply:
@jreback 
Without this, ci check failed... Does this imply that my modification is wrong?

Reply:
Do you remember the CI failure? It may have been a flaky test.

Reply:
@TomAugspurger 
Yes, the failure part is: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/series/test_constructors.py#L1352

Reply:
@jreback 

It does not raise an error and convert [ps] resolution, but,,, the value goes wrong.

```python
df = pd.DataFrame({'timestamp': [np.datetime64('2017-01-01 01:00:00'), np.datetime64('2017-01-01 02:00:00')],
                   'value': ['a', 'b']})
df['ps'] = df.loc[:, 'timestamp'].values.astype('datetime64[ps]')
```
```
            timestamp value                            ps
0 2017-01-01 01:00:00     a 1972-03-20 01:55:03.235870720
1 2017-01-01 02:00:00     b 1972-04-30 17:55:03.235870720
```